pkg/image/resolve_test.go

```diff
@@ -41,3 +48,186 @@ func TestConvertMirroredImages(t *testing.T) {
+		},
+		{
+			name: "Cluster empty URL, with default registry and rancher on image name",
+			input: input{
+				image:              "rancher/imagename",
+				CattleBaseRegistry: "default-registry.com",
+				cluster:            &v3.Cluster{},
+			},
+			expected: "default-registry.com/rancher/imagename",
+		},
+		{
+			name: "Cluster with URL, no default registry, no rancher/ on image",
+			input: input{
+				image:              "imagename",
+				CattleBaseRegistry: "",
+				cluster:            clusterWithPrivateRegistry("cluster-url.com"),
+			},
+			expected: "cluster-url.com/rancher/imagename",
+		},
+		{
+			name: "Cluster with URL, no default registry, with rancher/ on image",
+			input: input{
+				image:              "imagename",
+				CattleBaseRegistry: "",
+				cluster:            clusterWithPrivateRegistry("cluster-url.com"),
+			},
+			expected: "cluster-url.com/rancher/imagename",
+		},
+		{
+			name: "Cluster with URL, and default registry, no rancher/ on image",
+			input: input{
+				image:              "imagename",
+				CattleBaseRegistry: "registry-url",
+				cluster:            clusterWithPrivateRegistry("cluster-url.com"),
+			},
+			expected: "cluster-url.com/rancher/imagename",
+		},
+		{
+			name: "Cluster with URL, and default registry, with rancher/ on image",
+			input: input{
+				image:              "rancher/imagename",
+				CattleBaseRegistry: "registry-url",
+				cluster:            clusterWithPrivateRegistry("cluster-url.com"),
+			},
+			expected: "cluster-url.com/rancher/imagename",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := settings.SystemDefaultRegistry.Set(tt.input.CattleBaseRegistry); err != nil {
+				t.Errorf("Failed to test TestResolveWithCluster(), unable to set SystemDefaultRegistry with the value: %v", err)
+			}
+			assertlib.Equalf(t, tt.expected, ResolveWithCluster(tt.input.image, tt.input.cluster), "ResolveWithCluster(%v, %v)", tt.input.image, tt.input.cluster)
+		})
+	}
+}
+
+func TestResolve(t *testing.T) {
+	if os.Getenv("CATTLE_BASE_REGISTRY") != "" {
+		fmt.Println("Skipping TestResolve. Can't run the tests with CATTLE_BASE_REGISTRY set")
+		return
+	}
+
+	type input struct {
+		image              string
+		CattleBaseRegistry string
+	}
+	tests := []struct {
+		name     string
+		input    input
+		expected string
+	}{
+		{
+			name: "No default",
+			input: input{
+				image:              "imagename",
+				CattleBaseRegistry: "",
+			},
+			expected: "imagename",
+		},
+		{
+			name: "Default without rancher",
+			input: input{
+				image:              "imagename",
+				CattleBaseRegistry: "default-registry.com",
+			},
+			expected: "default-registry.com/rancher/imagename",
+		},
+		{
+			name: "Default wit rancher",
+			input: input{
+				image:              "rancher/imagename",
+				CattleBaseRegistry: "default-registry.com",
+			},
+			expected: "default-registry.com/rancher/imagename",
+		},
+	}
+	for _, tt := range tests {
```

Comment:
We should probably (re)set `settings.SystemDefaultRegistry` to `""` before we enter the loop.

Reply:
The loop already calls the set to the empty value if needed. 

If CATTLE_BASE_REGISTRY is set I'm skipping the test. If the env is set we can't call the ".Set()".

The default value of the  SystemDefaultRegistry is the  env from CATTLE_BASE_REGISTRY, so it will always be "" on the test.